src/dbnode/ts/writes/write_batch.go

```diff
@@ -102,14 +121,29 @@ func (b *writeBatch) Reset(
 	batchSize int,
 	ns ident.ID,
 ) {
-	var writes []BatchWrite
+	// Preallocate slightly more when not using initialBatchSize.
+	preallocateBatchCap := int(float32(batchSize) * preallocateBatchCoeff)
+
 	if batchSize > cap(b.writes) {
-		writes = make([]BatchWrite, 0, batchSize)
+		batchCap := batchSize
+		if cap(b.writes) == 0 {
```

Comment:
Why do this if cap of writes is zero? 

I thought you’d always want to allocate 1.2x the size you actually need since next request will likely have a little more or a little less.

Reply:
The reason for this was to preserve the original behaviour when having `InitialBatchSize` set in config (so that if you have initial batch size of 128 in config, and a batch size of 200 arrives, exactly 200 would be allocated, and not 200*1.2).
I'll add an explicit `writeBatch.adaptiveSize` flag which will help me handle every case better.